src/components/CreatingOpenGameModal.tsx

```diff
@@ -0,0 +1,72 @@
+import _ from 'lodash';
+import React from 'react';
+
+import { Button, Modal, ModalHeader, ModalBody } from 'reactstrap';
+
+import web3Utils from 'web3-utils';
+import * as gameStates from '../redux/game/state';
+
+interface Props {
+  gameState: gameStates.GameState;
+  createOpenGame: (roundBuyIn: string) => void;
+  cancelOpenGame: () => void;
+}
+
+export default class CreatingOpenGameModal extends React.PureComponent<Props> {
+  buyinInput: any;
+
+  constructor(props) {
+    super(props);
+    this.buyinInput = React.createRef();
+    this.createOpenGameHandler = this.createOpenGameHandler.bind(this);
+  }
+
+  createOpenGameHandler(e) {
+    e.preventDefault();
+    const buyin = Number(this.buyinInput.current.value);
+
+    // TODO: disable button when input is empty
+    if (!buyin || Number.isNaN(buyin)) {
+      return;
+    }
+
+    this.props.createOpenGame(web3Utils.toWei(buyin.toString(), 'ether'));
+    this.buyinInput.current.value = '';
+  }
+
+  render() {
+    const visible = (this.props.gameState.name === gameStates.StateName.CreatingOpenGame);
```

Comment:
We might want to decouple this from gameState and just have a visible properly we pass in.

Reply:
That's a good call.